### DISCUSSION TITLE
Convert a .docx file through url or base64

### DISCUSSION THREAD

**Discussion by PPetterssonSO** [gotenberg/gotenberg#719] [Q&A]
Hi!
First of all, great, GREAT solution.
Ive been, as many other, struggling with pdf converts for a long long time but this solution seems to be working great!

Question:
Can i somehow convert a .docx file by suppling an url or a base64?
Ive tested the forms/libreoffice/convert  endpoint and that works great, but, i do not wish to load the .docx file from disk.
Cant seem to find anything in the documentation

Is this possible?

Regards
Pär

**Comment by gulien** [comment]
Hello @Nawksen! Thanks for the kind words, I appreciate ☺️

You’re actually not the first one to ask this question, and, in a JSON world, most people seem to have forgotten how to work with a form data request 😅

It all depends on how you create your form files on the client side (cURL, an HTTP library in your language, etc.). For instance, in the PHP library I wrote for Gotenberg, [there is a Stream object](https://github.com/gotenberg/gotenberg-php/blob/main/src/Stream.php) that I later use to create a form file.

**Reply by PPetterssonSO** [reply]
Hi!
Of cours!
Got it working, thanks so much!
